torch/lib/c10d/test/TCPStoreTest.cpp

```diff
@@ -1,26 +1,36 @@
 #include <c10d/test/StoreTestCommon.hpp>
 
+#include <gtest/gtest.h>
 #include <cstdlib>
 #include <iostream>
 #include <thread>
 
 #include <c10d/PrefixStore.hpp>
 #include <c10d/TCPStore.hpp>
 
-void testHelper(const std::string& prefix = "") {
+// Different ports for different tests.
+void testHelper(int port, const std::string& prefix = "") {
   const auto numThreads = 16;
   const auto numWorkers = numThreads + 1;
+
+  std::unique_ptr<c10d::TCPStore> serverTCPStore;
+  std::unique_ptr<c10d::PrefixStore> serverStore;
   // server store
-  c10d::TCPStore serverTCPStore("127.0.0.1", 29500, numWorkers, true);
-  c10d::PrefixStore serverStore(prefix, serverTCPStore);
+  auto serverThread =
+      std::thread([&serverTCPStore, &serverStore, &prefix, &numWorkers, &port] {
+        serverTCPStore = std::make_unique<c10d::TCPStore>(
+            "127.0.0.1", port, numWorkers, true, std::chrono::seconds(30));
+        serverStore =
+            std::make_unique<c10d::PrefixStore>(prefix, *serverTCPStore);
 
-  // Basic set/get on the server store
-  c10d::test::set(serverStore, "key0", "value0");
-  c10d::test::set(serverStore, "key1", "value1");
-  c10d::test::set(serverStore, "key2", "value2");
-  c10d::test::check(serverStore, "key0", "value0");
-  c10d::test::check(serverStore, "key1", "value1");
-  c10d::test::check(serverStore, "key2", "value2");
+        // Basic set/get on the server store
+        c10d::test::set(*serverStore, "key0", "value0");
+        c10d::test::set(*serverStore, "key1", "value1");
+        c10d::test::set(*serverStore, "key2", "value2");
+        c10d::test::check(*serverStore, "key0", "value0");
+        c10d::test::check(*serverStore, "key1", "value1");
+        c10d::test::check(*serverStore, "key2", "value2");
```

Comment:
This thread terminates once this has run. I suppose you're relying on the instance and by extension the server thread to be kept alive by the `unique_ptr` that holds it?

I think that, alternatively, we should consider decouple starting the TCP server from the waiting for all workers. This logic is too complicated for what we should test here.

Reply:
> I'm not sure I followed this completely. The server TCP store is kept alive by the unique_ptr on line 16.

Understood.

> IIUC, is your suggestion to change the TCPStore constructor to not wait for workers and have a separate method for this?

Yes, I think this would a) simplify this test (and similar patterns), because you no longer need a thread to launch a thread, and b) it would allow for letting the server pick a free port automatically (now you have to wait for all workers to connect, so it can't be ambiguous).